src/api.ts

```diff
@@ -288,6 +290,16 @@ class RippleAPI extends EventEmitter {
     return results
   }
 
+  // @deprecated Use X-addresses instead
+  generateAddress(options?: GenerateAddressOptions): GeneratedAddress {
+    options.includeClassicAddress = true
```

Comment:
Generally, you want to avoid parameter mutability as well. Modifying this variable here will modify it in the enclosing scope as well, which might not be something the caller is expecting to happen. 

Do you think that a caller will ever want to manually set this option? If so, then make a copy of it before  modifying it. If not, then I would just remove this as an option on the `GenerateAddressOptions` object and pass an extra parameter into the generateAddressAPI.

Reply:
What about `return generateAddressAPI({...options, includeClassicAddress: true})`?

Reply:
Also, if options can be undefined you should give it a default value of empty object:

```
  generateAddress(options?: GenerateAddressOptions= {}): GeneratedAddress {
```

Reply:
+1 on these changes